app/src/main/java/com/example/myprofile/presentation/ui/fragments/auth/sign_up/SignUpViewModel.kt

```diff
@@ -0,0 +1,93 @@
+package com.example.myprofile.presentation.ui.fragments.auth.sign_up
+
+import android.content.SharedPreferences
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.myprofile.data.repository.UserDataRepository
+import com.example.myprofile.data.model.UserResponse
+import com.example.myprofile.data.repository.repository_impl.UserRepositoryImpl
+import com.example.myprofile.domain.states.ApiState
+import com.example.myprofile.presentation.utils.Validation
+import com.example.myprofile.presentation.utils.ext.saveAutoLoginData
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+/**
+ * This class is responsible for managing the user registration logic in the application. It uses LiveData and StateFlow to monitor data and registration status.
+ */
+@HiltViewModel
+class SignUpViewModel @Inject constructor(
+    private val sP: SharedPreferences,
+    private val userDataRepository: UserDataRepository,
+    private val userRepositoryImpl: UserRepositoryImpl
+) : ViewModel() {
+
+    // LiveData objects for user input
+    val email = MutableLiveData<String>()
+    val password = MutableLiveData<String>()
+    val name = MutableLiveData<String>()
+    val phone = MutableLiveData<String>()
+
+    // Flag to check if the checkbox is checked
+    var isCheckBoxChecked: Boolean = false
+    private val editor by lazy { sP.edit() }
```

Comment:
You will get rid of this field after reworking api request order because it will become redundant.